doc/reference/ovn-internals.md

```diff
@@ -0,0 +1,168 @@
+The OVN underlay is configured and managed by OVN.
+
+### Logical router
+A logical router is a virtualized router.
+There is one per LXD OVN network.
+This handles layer 3 networking and additionally has associated NAT rules and security policies.
+
+### Logical switch
+Logical routers cannot be directly connected to OVS bridges; for this, we use a logical switch.
+There are two logical switches per LXD OVN network:
+
+- The external switch, which connects via logical switch port to a port on the logical router and to the provider OVS switch.
+- The internal switch, which connects via logical switch port to a port on the logical router and to the integration bridge.
+  This switch contains DHCP and IP allocation configuration.
+
+### Logical switch/router ports
+When you create a logical router or switch in OVN, it doesn't initially have any ports.
+You need to create ports and then link them.
+For example, the internal logical switch and the logical router for a LXD OVN network are connected by:
+
+1. Adding a logical router port to the logical router.
+1. Adding a logical switch port to the internal logical switch.
+1. Configuring the internal logical switch port as a router port and setting the logical router name.
+
+Some configuration is applied directly at port level.
+For example, in a LXD OVN network, IPv6 router advertisement settings are applied on the logical router port for the internal switch.
+This is by design. It allows OVN to push configuration down to the port level so that packets are handled as quickly as possible.
+
+### Port groups
+When a LXD OVN network is created, a port group will be created that is specific to that network.
+When instances are connected to the network, logical switch ports are created for them on the internal switch.
+These logical switch ports are added to the port group for the network.
+When a port group is created or updated in the OVN northbound database, the address set table is automatically populated.
+Address sets are used for managing access control lists (ACLs).
+By creating and maintaining the port group, we can easily select the whole network when managing ACLs.
+
+## OVN Uplink
+An OVN network can specify an uplink network.
+That uplink network must be a managed network and be of type `physical` or `bridge`.
+From these managed network definitions LXD ascertains a `parent` interface to use for the uplink connectivity.
+
+For managed `bridge` networks the interface is the name of the network itself.
+
+For managed `physical` networks it is the per-cluster member value of the `parent` setting.
+The `parent` interface itself can have one of three types:
+
+- Linux native bridge.
+- OVS Bridge.
+- Physical interface (or bond).
+
+It is important to note that a `physical` managed network's `parent` interface can be any of these types, and that for a `managed` bridge network the parent interface can be either types of bridges.
```

Comment:
```suggestion
It is important to note that a `physical` managed network's `parent` interface can be any of these types, and that for a managed `bridge` network the `parent` interface can be either types of bridges.
```

Also a couple of quick questions. In a bridge network, does the chassis group only contain one chassis? And do we include the per-member `parent` interfaces as part of each OVN chassis?

Those are more for my own understanding, but if you think there are relevant differences on how we operate on chassis in each type of network feel free to add something in.

Reply:
I'm not sure I understand the question.

If LXD is running in standalone mode then there is only one chassis (the host) in the chassis group. If LXD is clustered then either all members are in the chassis group, OR only members with the `ovn-chassis` role are in the chassis group.

If there are multiple OVN networks, all OVN networks will contain the same cluster members in their chassis group. However, the chassis group for each network will assign different priorities to the chassis. This keeps uplink traffic distributed amongst the host machines.

Conceptually, the `parent` network is not something that is passed into OVN. We use an OVS bridge to get from the physical network to the virtual one and vice versa.